x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportPutDataFrameTransformAction.java

```diff
@@ -202,41 +197,12 @@ private void putDataFrame(DataFrameTransformConfig config, ActionListener<Respon
         // <5> Return the listener, or clean up destination index on failure.
         ActionListener<Boolean> putTransformConfigurationListener = ActionListener.wrap(
             putTransformConfigurationResult -> listener.onResponse(new Response(true)),
-            putTransformConfigurationException ->
```

Comment:
Since we no longer always create the destination, we consequently should not assume absolute ownership over it. So, deleting the index on a failure on `PUT` may actually delete an index we did not create. Additionally, this does not make sense since `PUT` no longer creates the destination index anyways.